src/agent/stacktrace-parser/src/lib.rs

```diff
@@ -0,0 +1,432 @@
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub module_path: Option<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub module_offset: Option<u64>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct CrashLog {
+    pub text: String,
+    pub sanitizer: String,
+    pub summary: String,
+    pub fault_type: String,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub call_stack: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub full_stack_details: Vec<StackEntry>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub full_stack_names: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub minimized_stack_details: Vec<StackEntry>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub minimized_stack: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub minimized_stack_function_names: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub scariness_score: Option<u32>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub scariness_description: Option<String>,
+}
+
+fn function_without_args(func: &str) -> String {
+    // TODO: This is an extremely niave approach to splitting arguments.k It
+    // doesn't handle C++ well.  As an example:
+    //
+    // This turns this:
+    // "base::internal::RunnableAdapter<void (__cdecl*)(scoped_ptr<blink::WebTaskRunner::Task,std::default_delete<blink::WebTaskRunner::Task> >)>::Run",
+    //
+    // Into this:
+    // "base::internal::RunnableAdapter<void "
+    //
+    // However, given this is intended to enable de-duplicating crash reports
+    // with ClusterFuzz, this is going to stay this way for now. This is used to
+    // fill in `minimized_stack_functions_names`. The unstripped version will be
+    // in `minimized_stack`.
+    let split: Vec<_> = func.splitn(2, '(').collect();
+    split[0].to_string()
+}
+
+fn filter_funcs(entry: &StackEntry, stack_filter: &RegexSet) -> Option<StackEntry> {
+    let mut entry = entry.clone();
+    if let Some(name) = &entry.function_name {
+        // mirror Clusterfuzz's replacing LLVMFuzzerTestOneInput
+        // with the fuzzer filename
+        //
+        // Ref: https://github.com/google/clusterfuzz/blob/
+        //    a6bb73e4988f4a7064e990a0a78cc6bc812ef741/src/python/
+        //    lib/clusterfuzz/stacktraces/__init__.py#L1362-L1373
+        if name == "LLVMFuzzerTestOneInput" {
+            if let Some(file_name) = &entry.source_file_name {
+                entry.function_name = Some(file_name.to_string());
+                return Some(entry);
+            }
+        }
+        if stack_filter.is_match(name) {
```

Comment:
Noting that by having this match come after the above, we disallow filtering out `LLVMFuzzerTestOneInput`, but we no doubt want that.

Reply:
Yep.  LLVMFuzzerTestOneInput is functions stripped out in the STACK_FRAME_IGNORE_REGEXES.   Given how not useful that is in the context of "hundreds of libfuzzer targets in this project", I can understand that desire.